google_maps/public/js/src/google_maps_edit.js

```diff
@@ -47,10 +71,23 @@ function GoogleMapsEditXBlock(runtime, element) {
     clearButton.prop('hidden', true);
   }
 
-  $(document).ready(function() {
+  /**
+   * Checks if the value was selected from Google Maps autocomplete dropdown
+   */
+  function validatePlaceNameField() {
+    if (iframePlaceNameInput.val() === googleMapsXBlockApp.place.name ) {
+      googleMapsXBlockApp.toggleErrorMessage($('#place-name-input'), false);
+      saveXBlockData();
+    } else {
+      googleMapsXBlockApp.toggleErrorMessage($('#place-name-input'), true);
+    }
+  }
+
+  $(function() {
     displayNameClearButton.prop('hidden', displayNameInput.val() === displayNameDefaultValue);
     iframeWidthClearButton.prop('hidden', iframeWidthInput.val() === iframeWidthDefaultValue);
     iframeHeightClearButton.prop('hidden', iframeHeightInput.val() === iframeHeightDefaultValue);
+    iframePlaceNameClearButton.prop('hidden', iframePlaceNameInput.val() === iframePlaceNameDefaultValue);
   });
```

Comment:
I didn't find a way to catch the exception. I don't know if this is necessary because google will return a message what happened and that message will get displayed instead of the map.